### PR TITLE
fix: chore: Migrate gateway auth token to SecretRef (#278)

### LINKED ISSUE #278: chore: Migrate gateway auth token to SecretRef
## Background
OpenClaw v2026.3.7 added SecretRef support for `gateway.auth.token`. Currently our token is stored as plaintext in `openclaw.json`.

## Proposed Change
Migrate `gateway.auth.token` to use a SecretRef (environment variable or file-backed).

## Benefits
- Token not visible in config dumps or version-controlled files
- Aligns with security best practices

## Acceptance Criteria
- [ ] Gateway auth token migrated to env var or file SecretRef
- [ ] Gateway starts and authenticates correctly
- [ ] `config.get` no longer shows plaintext token

### PR DESCRIPTION
**Pull request by markus-lassfolk**
## Summary

- Add `gateway.auth.token` field to plugin config with SecretRef support: `env:VAR_NAME` resolves from environment, `file:/path` reads from a file
- Resolved token stored as non-enumerable `_resolvedToken` — never visible in `JSON.stringify` / `config.get` output
- `init-databases.ts` now prefers `cfg.gateway.auth.token` (SecretRef-resolved) over the `OPENCLAW_GATEWAY_TOKEN` env var, with graceful fallback
- 23 new tests covering all SecretRef patterns and edge cases

Closes #278

## Test plan

- [x] `npm test` — all 2904 tests pass (118 test files)
- [x] `npx tsc --noEmit` — no TypeScript errors
- [x] New `tests/gateway-secret-ref.test.ts` (23 tests) covers `resolveSecretRef` and `parseGatewayConfig`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Changes how the gateway auth token is sourced and adds fail-closed behavior when a configured SecretRef can’t be resolved, which can break gateway-routed OpenAI calls if misconfigured. Touches auth/credential handling and startup initialization, but is scoped and covered by new tests.
> 
> **Overview**
> Adds `gateway.auth.token` to the memory-hybrid plugin config with SecretRef support (`env:`/`file:`), resolving the secret at startup while keeping only the reference string visible in config dumps via a non-enumerable `_resolvedToken`.
> 
> Updates OpenAI client initialization to prefer the config-provided, resolved gateway token over `OPENCLAW_GATEWAY_TOKEN`, and **throws** (no env fallback) if `gateway.auth.token` is set but cannot be resolved; also tightens env-key resolution trimming and updates the warning message. Includes a new Vitest suite covering SecretRef resolution and non-leakage in `JSON.stringify`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit db588ee9cbf8d822106c6dd9f0ab38ed4dac0a6e. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: extensions/memory-hybrid/setup/init-databases.ts [hunk 1]
```diff
@@ -144,15 +144,22 @@ function buildMultiProviderOpenAI(cfg: HybridMemoryConfig, api: ClawdbotPluginAp
   };
   const gatewayPortRaw = process.env.OPENCLAW_GATEWAY_PORT;
   const gatewayPort = gatewayPortRaw ? Number.parseInt(gatewayPortRaw, 10) : undefined;
-  const gatewayToken = process.env.OPENCLAW_GATEWAY_TOKEN;
+  // Resolve gateway auth token: prefer cfg.gateway.auth.token (SecretRef) over env var fallback.
+  // The parser stores the resolved value as non-enumerable _resolvedToken so it never appears in
+  // JSON dumps while remaining accessible here at runtime.
+  const gatewayAuthResolved = (cfg.gateway?.auth as Record<string, unknown> | undefined)?._resolvedToken as string | undefined;
+  const gatewayToken = gatewayAuthResolved ?? process.env.OPENCLAW_GATEWAY_TOKEN;
```
Comment (markus-lassfolk) [inline extensions/memory-hybrid/setup/init-databases.ts]: ### 🟡 Broken SecretRef falls back to stale env token

**Warning**

Once `gateway.auth.token` is configured, a resolution failure still falls back to `OPENCLAW_GATEWAY_TOKEN`. That makes rollout mistakes hard to detect: an unreadable or missing secret file can silently keep using an old env token even though the config claims the plugin-level secret should take precedence. I would fail closed here, or raise an explicit startup/config error, when a configured SecretRef cannot be resolved instead of transparently authenticating with a different credential.

File: extensions/memory-hybrid/setup/init-databases.ts [hunk 2]
```diff
@@ -144,15 +144,22 @@ function buildMultiProviderOpenAI(cfg: HybridMemoryConfig, api: ClawdbotPluginAp
   };
   const gatewayPortRaw = process.env.OPENCLAW_GATEWAY_PORT;
   const gatewayPort = gatewayPortRaw ? Number.parseInt(gatewayPortRaw, 10) : undefined;
-  const gatewayToken = process.env.OPENCLAW_GATEWAY_TOKEN;
+  // Resolve gateway auth token: prefer cfg.gateway.auth.token (SecretRef) over env var fallback.
+  // The parser stores the resolved value as non-enumerable _resolvedToken so it never appears in
+  // JSON dumps while remaining accessible here at runtime.
+  const gatewayAuthResolved = (cfg.gateway?.auth as Record<string, unknown> | undefined)?._resolvedToken as string | undefined;
```
Comment (markus-lassfolk) [inline extensions/memory-hybrid/setup/init-databases.ts]: ### 🟢 Give the hidden resolved token a real TypeScript type

**Suggestion**

`_resolvedToken` is currently threaded through the system via `Record<string, unknown>` casts. That works, but it gives up the type-safety this PR is otherwise trying to improve. Consider introducing an internal parsed type such as `GatewayAuthConfig & { readonly _resolvedToken?: string }` (or a dedicated symbol-backed field) so consumers do not need unchecked casts and refactors cannot accidentally misspell the hidden property.

**Review by markus-lassfolk — COMMENTED** [review]
## 🤖 GPT Council Review — PR #322
Summary: 0 critical, 1 warning, 1 suggestion.
Signed: 🤖 GPT Council Member
